### DISCUSSION TITLE
How to extend vscode search?

### DISCUSSION THREAD

**Discussion by a7madgamal** [microsoft/vscode-discussions#13] [Extension Development QnA] [answered]
I want to create a search plugin that works/looks exactly like a normal search in vscode + more options based on the results
Basically, I want to use the language server (typescript for example) to:

1. Find the AST node type of each result to know its context (comment, import statement, preserved keyword...)
2. Based on all the node types in this result set: show labels to include/exclude this node type from the list

I don't have a specific question yet but any help will be much appreciated.

**Comment by alefragnani** [comment, marked as answer]
Hi @a7madgamal ,

There is no API for that, so you can't _customize_ the Search feature.

Maybe the **References View** extension (https://marketplace.visualstudio.com/items?itemName=ms-vscode.references-view) is something that could be useful for you. But as you can see, it does not _injects_ anything to the Search feature but instead, creates its own results view.

Hope this helps
